mne/preprocessing/_css.py

```diff
@@ -0,0 +1,174 @@
+
+
+import numpy as np
+import mne
+import matplotlib.pyplot as plt
+
+
+def _css(mag_data, grad_data, filter_data, r=6):
+    """Remove the cortical contribution to filter_data (typically
+    magnetometer or EEG data) by means of a temporal projection.
+
+    This procedure removes the common signal subspace between mag_data
+    and grad_data from filter_data using r number of projection vectors.
+    Note that the SNR needs to be relatively high for this to work; it
+    therefore makes most sense using this on averaged data in an evoked
+    object.
+
+    Parameters
+    ----------
+    mag_data : np.ndarray of float, shape (n_mag_sensors,n_times)
+        The magnetometer data. Can use either all magnetometer data or
+        a few selected sensors close to a region you want to suppress.
+    grad_data : np.ndarray of float, shape (n_grad_sensors,n_times)
+        The gradiometer data. Can use either all gradiometer data or
+        a few selected sensors close to a region you want to suppress.
+    filter_data : np.ndarray of float, shape (n_grad_sensors,n_times)
+        The data to be filtered, typically the magnetometer or EEG data.
+    r : int
+        The number of projection vectors.
+
+    Returns
+    -------
+    filtered_data : np.ndarray of int, shape (n_features)
+        The outlier indices.
+    """
+    # Orthonormalize gradiometer and magnetometer data by a QR decomposition
+    Qg, rg = np.linalg.qr(grad_data.T)
+    Qm, rm = np.linalg.qr(mag_data.T)
+
+    # Calculate cross-correlation
+    C = np.dot(Qg.T, Qm)
+
+    # Channel weights for common temporal subspace by SVD of cross-correlation
+    Y, S, Z = np.linalg.svd(C)
+
+    # Get temporal signals from channel weights
+    u = np.dot(Qg, Y)
+
+    # Project out common subspace (r is number of projection vectors)
+    filtered_data = filter_data
+    for i in range(r):
+        proj_vec = u[:, i].reshape(u.shape[0], 1)
+        NpProj = np.dot(filtered_data, proj_vec)
+        filtered_data = filtered_data - np.dot(NpProj, proj_vec.T)
+
+    return filtered_data
+
+
+def _plot_psd(evoked, evoked_subcortical, filt_inds, title):
+    """Plots the total power spectral density of data in evoked and
+    evoked_subcortical.
+
+    Plots the sum of the power spectral density across channels filt_inds.
+
+    Parameters
+    ----------
+    evoked : evoked object
+        Unprocessed data.
+    evoked_subcortical : evoked object
+        Processed data.
+    filt_inds : np.ndarray of int, shape (n_sensors)
```

Comment:
```suggestion
    filt_inds : array of int, shape (n_sensors,)
```

be use this consistent formatting in all your docstrings.